packages/ember-application/lib/system/application.js

```diff
@@ -762,6 +776,10 @@ const Application = Engine.extend({
       _loaded.application = undefined;
     }
 
+    if (this._applicationInstances.length) {
+      this._applicationInstances.forEach(i => i.destroy());
+    }
+
     if (this._globalsMode && this.__deprecatedInstance__) {
```

Comment:
Seems like this might be able to be removed now? `this.__deprecatedInstance__` is assigned to the result of calling `this.buildInstance()`, so the `this.__deprecatedInstance__` would already have been destroyed by the loop above...